### PR TITLE
backend: change backend dir

### PR DESCRIPTION
**Pull request by JonasVautherin**
* backend is now in /backend instead of /grpc/server
* update readme
* change BUILD_DRONECORESERVER to BUILD_BACKEND in Makefile

### CODE REVIEW

File: backend/README.md [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+```
+make BUILD_BACKEND=YES
+```
+
+### Run the backend
+
+The backend is built as an executable:
+
+```
+./build/default/backend/src/backend_bin
```
Comment (shakthi-prashanth-m) [inline backend/README.md]: Are you sure you're placing backend executable in `src` ?

File: backend/src/CMakeLists.txt [hunk 2]
```diff
@@ -45,7 +45,7 @@ target_include_directories(backend
     PRIVATE
     ${CMAKE_SOURCE_DIR}/core
     ${CMAKE_SOURCE_DIR}/plugins
-    ${CMAKE_BINARY_DIR}/grpc/server/src
+    ${CMAKE_BINARY_DIR}/backend/src
```
Comment (shakthi-prashanth-m) [inline backend/src/CMakeLists.txt]: I just noticed. Are you sure you're placing backend executable in `src` ?

File: backend/README.md (hunk 1 above)
Comment (shakthi-prashanth-m) [reply backend/README.md]: I am not sure the reason behind having it in `src`. As you know, for DroneCore unit tests, we've in  `build/default/unit_tests_runner`.  I think `./build/default/backend/bin` is also fine.

Comment (JonasVautherin) [reply backend/README.md]: Right. I can put them in `build/default`, of course.

Comment (JonasVautherin) [reply backend/README.md]: Actually, I am not sure what a good practice would be for that (and also maybe `install` should be set for the backend). I don't have much time for this right now, I just wanted to move `grpc/server` to `backend` now that not so many people are working on it.

Maybe changing the build output/install targets can be done later?

Comment (shakthi-prashanth-m) [reply backend/README.md]: Yeah its perfectly fine.

**Pull request merged**